tools/src/main/scala/scala/scalanative/codegen/llvm/compat/os/UnixCompat.scala

```diff
@@ -11,19 +11,15 @@ private[codegen] class UnixCompat(codegen: AbstractCodeGen)
 
   import codegen.{pointerType => ptrT}
 
-  val ehWrapperTy = "@_ZTIN11scalanative16ExceptionWrapperE"
   val excRecTy = s"{ $ptrT, i32 }"
-  val beginCatch = "@__cxa_begin_catch"
-  val endCatch = "@__cxa_end_catch"
+  val scalaNativeCatch = "@scalanative_catch"
   val catchSig =
-    if (useOpaquePointers) s"$ptrT $ehWrapperTy"
-    else s"i8* bitcast ({ i8*, i8*, i8* }* $ehWrapperTy to i8*)"
+    if (useOpaquePointers) s"$ptrT null"
+    else s"i8* null"
   val landingpad =
     s"landingpad $excRecTy catch $catchSig"
-  val typeid =
-    s"call i32 @llvm.eh.typeid.for($catchSig)"
 
-  protected val osPersonalityType: String = "@__gxx_personality_v0"
+  protected val osPersonalityType: String = "@scalanative_personality"
```

Comment:
I wonder if we should keep the original variant of `UnixCompat` for exception handling and either old or new variant depending on `Config.usingCPPExceptions`. This would also allow us to keep old behaviour of being able to catch C++ exception from code that is calling Scala Native library (probably)

Reply:
In my opinion, allowing to catch Scala Native exceptions from C++ is a feature nobody uses and it can't really be used, since the wrapped exception is a `void *` that has no meaning in C++, you can't get the class, the error message, etc. (you can only if you play with pointer arithmetics, and the object structure is private to the Scala Native runtime).
We could do that, but I don't see value in it, and having the code running in both modes it's simpler and would lead to fewer surprises when using `c++` code.

Reply:
I am a little confused on the idea of catching C++ exceptions at all. I see three use cases, but maybe there are more.

1. The user puts C++ glue code but needs have `extern C` to allow Scala Native to call it.
2. A library uses C++ but must have `extern C` to allow Scala Native to call it.
3. Create a Scala Native library. It must have exposed C functions to call it.

I don't see any reason to support C++ exceptions for the above because we only interact with the C ABI and thus standard C error handling techniques would be used. I don't think throwing a C++ exception across those boundaries makes sense.

As I recall, using C++ and `libunwind` was just used as the easiest way to provide for 1) Java/Scala exceptions and 2) stack traces. I could be off base but if we can support these two features and the cases above in a portable manner then we should be good. Detecting to use C++ to link is the trick but would that mean that a `.cpp` file was encountered? - I think that was mentioned that they can't be empty for the conditional compilation case?

We would just keep the status quo for Windows or would it be possible the support PE or other binary formats that might come up similar to the Dwarf support? Hopefully this discussion is useful and not completely off base.

Reply:
One use case would be to provide a catch all clause in C++ when calling a Scala Native library.
It's an advanced use case which can be also solved by returning a Either-like data structure in C.
As I said earlier, I think this is more a bug than a feature, I think the best behavior is just to crash.

Reply:
Calling a C ABI and expecting a C++ exception is a bit much. If I was calling from C I wouldn't expect to get a C++ exception so I think that would be a crash. I agree.